Zotero/Scenes/General/Views/ContainerViewController.swift

```diff
@@ -156,8 +145,19 @@ class ContainerViewController: UIViewController {
     }
 
     // MARK: - Actions
-
     @objc private func hide() {
         presentingViewController?.dismiss(animated: true, completion: nil)
     }
+
+    private func moveToKeyboard(_ data: KeyboardData, willShow: Bool) {
```

Comment:
I guess we should decide on a rule here. If I understand correctly you moved this away from `viewDidLoad` because it was capturing `self`? 

So if we want to stick to our rule, that if a function is used only in one place, it should be moved to that place as child function, this should be moved back and you should add `self: ContainerViewController` parameter and use explicit `self` in that child function.

Or we want to make an exception to our rule so that this case is simpler. Then we'll have some one-use standalone functions as this one while others are "hidden" as child functions.

I don't really know which is best. We already make an exception for `update(state: SomeState)` functions because otherwise we'd have the whole view controller included in a `viewDidLoad`. So that makes sense to me. I'm not really sure with this one. I guess I'm slightly more inclined to include `self` as parameter to keep it a bit cleaner. But then we'll have explicit `self`s in some functions (which we already have to have in some places anyway).

Thoughts? :)

Reply:
I guess we should just discuss these cases and see if we can come with a list of rules/suggestions to follow.
Indeed `update(state: SomeState)` is a very clear and cut case.

Here the issue is the indirect capture of `self` in the notification blocks, as the nested function can access the context of its parent function. 

So the rule to not rely on the context, but pass directly what is needed, forces as to be more aware. But still, it is easy to write the name of a class property without thinking about it, and thus cause `self` capture (granted, one of the pitfalls of not having to prefix `self.` explicitly).

The counter argument rule could be to not nest any function called by an escaped block, so that we are certain there are no indirect captures, but not all such cases are problematic.

I'd say let's go with the first one for the time being, and re-evaluate if it becomes too cumbersome. I'm guessing that this is not a common issue anyway.

Check https://github.com/zotero/zotero-ios/pull/831/commits/1758c9ef6bd22d7fef78216fa5b37dd4b487361c